src/blob/index/core.rs

```diff
@@ -296,27 +315,41 @@ where
     }
 
     async fn get_all(&self, key: &K) -> Result<Option<Vec<RecordHeader>>> {
-        match &self.inner {
+        let res = match &self.inner {
             State::InMemory(headers) => Ok(headers.get(key).cloned()),
             State::OnDisk(findex) => findex.find_by_key(key).await,
-        }
+        };
+        res.map(|r| {
+            r.and_then(|mut hs| {
+                hs.sort_by_key(|h| h.created());
+                hs.iter().fold(None, |s, n| {
```

Comment:
This is not optimal. It is better to find the last `None` item and remove it and everything before directly from the vector without recreating it.